### DISCUSSION TITLE
Does fzf-lua run in a Neovim terminal window?

### DISCUSSION THREAD

**Discussion by louwers** [ibhagwan/fzf-lua#1505] [General]
In [LazyVim for Ambitious Developers](https://lazyvim-ambitious-devs.phillips.codes/course/chapter-4/) this is said:

> There are a couple downsides to Fzf.lua, though. The main one is that it secretly runs in a Neovim terminal window

Is that true? Because I cannot find anything in the source code that would indicate that.

**Comment by ibhagwan** [comment]
> There are a couple downsides to Fzf.lua, though. The main one is that it secretly runs in a Neovim terminal window, so Normal mode behaves really weird.

Reading the link provided extra context, I now understand what the author means, correct fzf-lua is a terminal window as it’s a terminal utility, it is no secret, as such, neovim terminal “normal” mode is different and less usable, I personally don’t find this an issue as I don’t need “normal” window and use fzf finds for common operations (first, last, next page, scroll preview, etc). If you’re heavily using Normal mode in Telescope this might be an issue for you, that said, in 3 years+ of existence of fzf-lua it has never been an issue for users.


Overall it seems the author of the link you provided is somewhat biased, they never really user fzf-lua properly and the writing is lacking accordingly.


> Because I cannot find anything in the source code that would indicate that.

As for this bit:
https://github.com/ibhagwan/fzf-lua/blob/ce1e24f2a48089b65deefcb0a71a9856839002b5/lua/fzf-lua/fzf.lua#L222-L236

**Reply by louwers** [reply]
Thank you! I am also not interested in using normal mode in this context. Yes the author is biased, but I wanted to make sure this is at least correct.

**Reply by ibhagwan** [reply]
> Thank you! I am also not interested in using normal mode in this context. Yes the author is biased, but I wanted to make sure this is at least correct.

My best advice is just give it a spin, if you’re using mostly Telescope defaults use fzf-lua’s telescope profile (in setup or use `:FzfLua profiles`) and just use it for a while, unless you’re using some obscure extensions you’re not going to miss any functionality, after a while you’ll feel/see the differences so you can make a more informed decision which plug-in suits you more.

**Reply by louwers** [reply]
Oh I am a fzf-lua convert already. 🙂 

And the performance is miles ahead of Telescope in my experience.
